goal_src/levels/darkcave/darkcave-obs.gc

```diff
@@ -5,33 +5,425 @@
 ;; name in dgo: darkcave-obs
 ;; dgos: DAR, L1
 
-;; TODO - for cavecrystal-light
+(define-extern *cavecrystal-sg* skeleton-group)
+
+(declare-type cavecrystal process-drawable)
+
+(define-extern cavecrystal-idle (state cavecrystal))
+(define-extern cavecrystal-active (state cavecrystal))
+
+;; TODO - have to fully forward declare this,
+;; > Type Error: Type cavecrystal-light-control is not fully defined.
+;; > (execute-connections *cavecrystal-light-control*)
+(deftype cavecrystal-light (structure)
```

Comment:
calling this out, maybe it can be avoided like before

Reply:
Good idea but then this causes problems in other files like `cavecrystal-light` where it expects darkcave to be built first :(